src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs

```diff
@@ -450,9 +452,21 @@ public IEnumerable<PackageSource> LoadNuGetSources(PackageId packageId, PackageS
             throw new NuGetPackageInstallerException("No NuGet sources are defined or enabled");
         }
 
+        CheckHttpSources(sources);
         return sources;
     }
 
+    private void CheckHttpSources(IEnumerable<PackageSource> packageSources)
+    {
+        foreach (var packageSource in packageSources)
+        {
+            if (packageSource.IsHttp && !packageSource.IsHttps && !packageSource.AllowInsecureConnections)
```

Comment:
This is the correct check as https is also considered http: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSource.cs#L49